pdns/dnsdistdist/doh.cc

```diff
@@ -699,7 +699,7 @@ static void processDOHQuery(DOHUnitUniquePtr&& du)
 
     if (du->downstream->d_config.useProxyProtocol) {
       size_t payloadSize = 0;
-      if (addProxyProtocol(dq)) {
+      if (addProxyProtocol(dq, &payloadSize)) {
```

Comment:
There is one `addProxyProtocol()` call left without the size pointer argument in `dnsdist.cc`. It's hard to decide for me if it needs the 2nd argument there. Maybe make it a required argument to force the developer to think if the size should be handled?

Reply:
It's not needed for the call in `dnsdist.cc`, as the query is sent over UDP. We only need to keep the size of the payload around in the other cases because we have to prepend the DNS payload with its size over TCP, and thus we need to know where to insert it (after the proxy protocol one).
Making the size a required argument (so probably a reference) would in my mind convey a different intent: that you _have_ to use the size, which is not true.

Reply:
Fair enough.